### PR TITLE
10-final Dmytro S.

### PR DESCRIPTION
**Pull request by dmytro-shevchenko1**
https://dmytro-shevchenko1.github.io/fe-course-basic-2023/

### CODE REVIEW

File: docs/dmytro-shevchenko1/GamesProject/calculator.html [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        <link rel="stylesheet" type="text/css" href="styles/footer.css">
+        <title>Calculator</title>
+    </head>
+    <body class="calculator-page">
+        <header class="header">
+            <div class="container">
+                <div class="header-flex">
+                    <img src="images/logo.svg" alt="Logo">
+                    <div class="nav">
+                        <ul class="nav-items">
```
Comment (vladyslav-yermolin-moc) [inline docs/dmytro-shevchenko1/GamesProject/calculator.html:20]: Hey, @dmytro-shevchenko1, I see a content jumps for different values
image
image

Comment (vladyslav-yermolin-moc) [reply docs/dmytro-shevchenko1/GamesProject/calculator.html:20]: And by the design, the calculator should be aligned to the left
image

File: docs/dmytro-shevchenko1/GamesProject/calculator.html [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                    </select>
+                </label>
+                <label>
+                    <input id="secondValue">
+                </label>
+                <button class="result-button" id="calculate" type="button">=</button>
+                <span class="result-text" id="result"></span>
+            </div>
+        </main>
+        <footer class="footer">
```
Comment (vladyslav-yermolin-moc) [inline docs/dmytro-shevchenko1/GamesProject/calculator.html:60]: Don't you like to make your footer [sticky](https://developer.mozilla.org/en-US/docs/Web/CSS/Layout_cookbook/Sticky_footers)?
image

File: docs/dmytro-shevchenko1/GamesProject/scripts/games.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    const platformSelect = document.querySelector('[data-type-filter="platform"]');
+    const searchInput = document.querySelector('[data-type-filter="search"]');
+
+    filterForm.addEventListener('change', filterGames);
+    newCheckbox.addEventListener('change', filterGames);
+    oldCheckbox.addEventListener('change', filterGames);
+    genreSelect.addEventListener('change', filterGames);
+    platformSelect.addEventListener('change', filterGames);
+
+    searchInput.addEventListener('keydown', (event) => {
```
Comment (vladyslav-yermolin-moc) [inline docs/dmytro-shevchenko1/GamesProject/scripts/games.js:133]: 👍

File: docs/dmytro-shevchenko1/GamesProject/scripts/games.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        games = await response.json();
+        renderCards(gameList, games.slice(0, 50));
+    } catch (error) {
+        console.error('GET error:', error);
+    } finally {
+        hideLoader();
+    }
+}
+
+document.addEventListener('DOMContentLoaded', fetchDataCards);
```
Comment (vladyslav-yermolin-moc) [inline docs/dmytro-shevchenko1/GamesProject/scripts/games.js:85]: This logic relates to initialization of the page, so, proly you can move it to `init` function. Or even do better:
```js
// at the bottom of the script
function attachHandlers() {
// lines 120-140
}

function init() {
  attachHandlers();
  fetchDataCards();
}

document.addEventListener('DOMContentLoaded', init);
```

File: docs/dmytro-shevchenko1/GamesProject/styles/cards.css [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+
+.cards {
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: center;
+  gap: 25px;
+  padding: 0;
+}
+
+ul.cards {
```
Comment (vladyslav-yermolin-moc) [inline docs/dmytro-shevchenko1/GamesProject/styles/cards.css:76]: Why do you need it? You can move `margin: 0;` into `.cards` selector, on line 68

File: docs/dmytro-shevchenko1/GamesProject/styles/calculator.css [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+
+.field {
+  display: flex;
+  align-items: center;
+  column-gap: 10px;
+  margin: 60px;
+}
+
+input {
```
Comment (vladyslav-yermolin-moc) [inline docs/dmytro-shevchenko1/GamesProject/styles/calculator.css:22]: Please add class
> Задавайте стилі лише на класи (не на елементи)

**Review by vladyslav-yermolin-moc — APPROVED** [review]
Good work!

**Review by vladyslav-yermolin-moc — CHANGES_REQUESTED** [review]
Sorry for the misclick, previously I've approved your pr by mistake
Don't forget to resolve the conflicts
image

File: docs/dmytro-shevchenko1/GamesProject/scripts/games.js (hunk 4 above)
Comment (dmytro-shevchenko1) [reply docs/dmytro-shevchenko1/GamesProject/scripts/games.js]: thanks for advice, that's a great improve for a code!

**Comment by dmytro-shevchenko1** [comment]
Hi, @vladyslav-yermolin-moc, thank you for the review. I fixed some things, and I hope I resolved the conflicts in the right way...

File: docs/dmytro-shevchenko1/GamesProject/styles/calculator.css [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+.field {
+  display: flex;
+  align-items: center;
+  column-gap: 10px;
+  margin-top: 60px;
+}
+
+.calc-input {
+  border: .006rem solid #A9A9A9;
```
Comment (vladyslav-yermolin-moc) [inline docs/dmytro-shevchenko1/GamesProject/styles/calculator.css:20]: Could you state in in `px`?

File: docs/dmytro-shevchenko1/GamesProject/styles/style.css [hunk 8]
```diff
@@ -0,0 +1,13 @@
+@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap");
+
+body {
```
Comment (vladyslav-yermolin-moc) [inline docs/dmytro-shevchenko1/GamesProject/styles/style.css:3]: I think you should make a `body` have the min height of the screen and flex column container. And then make `main` flex-growing to get all free place. 
Now you have a lot of empty spaces between the main content and the footer
image

**Review by vladyslav-yermolin-moc — APPROVED** [review]
Ok, fix with the next updates.

**Review by MZabolotnev — APPROVED** [review]

**Comment by dmytro-shevchenko1** [comment]
Hi, @vladyslav-yermolin-moc, @MZabolotnev, I added recent updates and that commit dissmised your approves :( I hope it's ok.

**Review by vladyslav-yermolin-moc — APPROVED** [review]

**Pull request merged**